package/MDAnalysis/coordinates/LAMMPS.py

```diff
@@ -552,16 +579,51 @@ def _read_next_timestep(self):
 
         indices = np.zeros(self.n_atoms, dtype=int)
 
-        f.readline()  # ITEM ATOMS etc
-        for i in range(self.n_atoms):
-            idx, _, xs, ys, zs = f.readline().split()
+        atomline = f.readline()  # ITEM ATOMS etc
+        attrs = atomline.split()[2:]  # attributes on coordinate line
+        col_ids = {attr: i for i, attr in enumerate(attrs)}  # column numbers
+        keys = col_ids.keys()
+        # check for ids and what type of coordinate convention
+        ids = "id" in keys
+        # coord_dict keys = convention keys = column data or False
+        coord_dict = {conv: False for conv in self._conventions[1:]}
+        coord_dict['unscaled'] = self._get_column(
+            ["x", "y", "z"], col_ids)  # unscaled
+        coord_dict['scaled'] = self._get_column(
+            ["xs", "ys", "zs"], col_ids)  # scaled
+        coord_dict["unwrapped"] = self._get_column(
+            ["xu", "yu", "zu"], col_ids)  # unwrapped
+        coord_dict["scaled_unwrapped"] = self._get_column(
+            ["xsu", "ysu", "zsu"], col_ids)  # scaled unwrapped
+
+        # this should only trigger on first read of "ATOM" card, after which it
+        # is fixed to the guessed value. Auto proceeds unscaled -> scaled
+        # -> unwrapped -> scaled_unwrapped
+        if self.lammps_coordinate_convention == "auto":
+            match = [var for var in self._conventions[1:] if coord_dict[var]]
```

Comment:
If you order `COORDTYPE_COLUMN_NAMES` in the desired checking order, you could take the resulting `list(coord_dict)[0]` instead.

Reply:
self._conventions is ordered in the correct checking order as is. :)

Reply:
Yes, I just meant that it didn't need to exist if you use the dict I suggested.

Reply:
Ah I see.